From 42e85a73b23ea139b645d9b425497485d7df1028 Mon Sep 17 00:00:00 2001
From: Ren Chenglei <chenglei.ren@intel.com>
Date: Fri, 9 Jun 2023 21:44:27 +0530
Subject: [PATCH] INTERNAL: Fix CtsHardwareTestCases failures

Add BO_USE_SW_MASK usage to DRM_FORMAT_RGB888 format,
this is required by Android R CTS.

Test: android.hardware.cts.HardwareBufferTest#testCreate

Tracked-On: OAM-95726
Signed-off-by: Ren Chenglei <chenglei.ren@intel.com>
---
 i915.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/i915.c b/i915.c
index b6ca8a1..246c41f 100644
--- a/i915.c
+++ b/i915.c
@@ -267,6 +267,7 @@ static int i915_add_combinations(struct driver *drv)
 	drv_add_combination(drv, DRM_FORMAT_BGR888, &metadata_linear, BO_USE_SW_MASK);
 #ifdef USE_GRALLOC1
 	drv_modify_combination(drv, DRM_FORMAT_ABGR2101010, &metadata_linear, BO_USE_SW_MASK);
+	drv_add_combination(drv, DRM_FORMAT_RGB888, &metadata_linear, BO_USE_SW_MASK);
 #endif
 
 	/*
-- 
2.40.1

